### PR TITLE
feat(1860): common ui components for plugins

### PR DESCRIPTION
**Pull request by cduflo**

### CODE REVIEW

File: src/components/serviceList.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  );
+
+  // prefetch logos
+  useEffect(() => {
+    if (services) {
+      Object.values(services).forEach((service) => {
+        const img = new Image();
+        img.src = service.logo;
+      });
+    }
```
Comment (jackcloudquery) [inline src/components/serviceList.tsx:69]: Nice! But doe we need to have this `if`? Because according to the types it should always be defined

**Review by jackcloudquery — APPROVED** [review]
Great, just a couple minor things

File: src/components/serviceList.tsx (hunk 1 above)
Comment (cduflo) [reply src/components/serviceList.tsx]: great point! I think I had a second param in the dependency array earlier and there was a chance of an undefined services. but no longer 👍

File: src/components/renderGuide/codeSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      return '<span class="' + cls + '">' + match + '</span>';
+    },
+  );
+}
+
+interface Props {
+  text: string;
+}
+
+export function CodeSnippet({ text }: Props) {
```
[the remark this answers is not in the document]
Comment (cduflo) [reply src/components/renderGuide/codeSnippet.tsx:33]: it was originally just a json viewer, but it does fallback to just a code snippet generator if the json doesn't parse -- which enables things like: https://www.figma.com/design/jzf4HIHivdvXz8RHpkxsUN/Cloud---cloud.cloudquery.io?node-id=713-40863&m=dev

**Pull request merged**